library/common/http/client.cc

```diff
@@ -346,10 +352,8 @@ void Client::DirectStream::saveFinalStreamIntel() {
   setFromOptional(latency_info_.dns_end_ms,
                   request_decoder_->streamInfo().downstreamTiming().getValue(
                       "envoy.dynamic_forward_proxy.dns_end_ms"));
-  // TODO(alyssawilk) sort out why upstream info is problematic for cronvoy tests.
-  return;
   if (info.upstreamInfo().has_value()) {
-    latency_info_.upstream_info_ = request_decoder_->streamInfo().upstreamInfo();
+    callbacks_->setFinalStreamIntel(request_decoder_->streamInfo().upstreamInfo().get());
```

Comment:
So my goal here was to avoid saving all the many intel fields on every call to saveLatestStreamIntel (which is called often) but instead to latch the shared pointer, and only do all the work once at the end of the stream.
If that simply doesn't work because of the cronvoy lifetime (which I'm still perplexed and dismayed about) I think it's redundant to setFinalStreamIntel here, and before we pass the intel up, and we can remove those call sites and latency_info_.

Reply:
Cronet updates the latest `sent_byte_count` on every callback, which is wha the code is doing here.

On the other hand, that value is only available in FinalStreamIntel, which obviously defeats the purpose of updating ip over each callback.

I will have to take another look at this logic.